### DISCUSSION TITLE
How to best add own information to assembler hover text?

### DISCUSSION THREAD

**Discussion by PlasmaHH** [compiler-explorer/compiler-explorer#6787] [Q&A]
I have added me another compiler to the C++ compilers for our local instance. The output assembler mnemonics sometimes differ from the usual ones that the compiler explorer knows about.

Also for some of the existing ones I would like to add some additional information.

Now the source of the info seems to be something like lib/asm-docs/generated/asm-docs-arm32.ts but the path suggests that this is generated information and it seems like it is best to not interfere there (to not have problems with future updates).

Is there any extra file or hook or similar where I could add information (preferrably that would be first looked into so it can override the shipped data)?

**Comment by mattgodbolt** [comment]
There's not an obvious hook: but you could replace the generated files on your local instance and/or hack the code that calls it to check your own ASM first -- `lib/handlers/assembly-documentation.ts` is probably the best place to start there.

**Reply by PlasmaHH** [reply]
Yeah I thought about hacking it somewhere into it, however that always makes upgrading a pain. Especially for such information it would really be useful like in many other programs if it was possible to put input data at multiple places.
